src/network/networkd-ndisc.c

```diff
@@ -172,12 +240,63 @@ static int ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
         return 0;
 }
 
+static int ndisc_router_generate_address(Link *link, unsigned prefixlen, uint32_t lifetime_preferred, Address *address) {
+        bool prefix_stable = false;
+        struct in6_addr addr;
+        IPv6Token *j;
+        Iterator i;
+        int r;
+
+        assert(address);
+        assert(link);
+
+        addr = address->in_addr.in6;
+        ORDERED_HASHMAP_FOREACH(j, link->network->ipv6_tokens, i) {
+                if (j->address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE
+                    && memcmp(&j->prefix, &addr, FAMILY_ADDRESS_SIZE(address->family)) == 0) {
+                        for (; j->dad_counter < DAD_CONFLICTS_IDGEN_RETRIES_RFC7217; j->dad_counter++) {
+                                r = ipv6_makestableprivate(link, &addr, prefixlen, &j->dad_counter, &address->in_addr.in6);
+                                if (r < 0)
+                                        return r;
+
+                                if (!ipv6_interface_identifiers(&address->in_addr.in6)) {
+                                        prefix_stable = true;
+                                        break;
+                                }
+                        }
+                } else if (j->address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_EUI64) {
+                        if (memcmp(&j->prefix, &address->in_addr, FAMILY_ADDRESS_SIZE(address->family)) == 0) {
+                                memcpy(((char *)&address->in_addr.in6) + 8, ((char *)&j->prefix) + 8, 8);
```

Comment:
I cannot understand this part. If two address is equivalent, then copy??? Also, this part seems to break backward compatibility.